### PR TITLE
Test_array_job_email.test_emails() fails to read email file

### PR DESCRIPTION
**Pull request by latha-subramanian**
#### Bug/feature Description
Test_array_job_email.test_emails fails to read the mail file (/var/mail/pbsuser).

#### Affected Platform(s)
* All Linux

#### Cause / Analysis / Design
There is no sudo privilege to open the /var/mail/pbsuser file

#### Solution Description
Use PTL's open_log() function in PBSLogUtils() with argument 'sudo=True' so the file is accessed as a privileged user.

#### Testing logs/output
[beforefix.txt](https://github.com/PBSPro/pbspro/files/2587246/beforefix.txt)
[afterfix.txt](https://github.com/PBSPro/pbspro/files/2587249/afterfix.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by latha-subramanian** [comment]
Thanks @vstumpf !
As far as I could see, this interface was most appropriate in PTL.

**Comment by latha-subramanian** [comment]
@PBSPro/pbspro-maintainers , please review.

**Comment by latha-subramanian** [comment]
Thanks @mike0042.
I have squashed and rebased. Please take this PR forward.

File: test/tests/functional/pbs_array_job_mail.py [hunk 1]
```diff
@@ -79,7 +80,7 @@ def test_emails(self):
         self.logger.info("Wait 10s for saving the e-mails")
         time.sleep(10)
 
-        with open(mailfile, 'r') as f:
+        with PBSLogUtils().open_log(log=mailfile, sudo=True) as f:
```
Comment (bhroam) [inline test/tests/functional/pbs_array_job_mail.py]: Use self.du.cat() instead of PBSLogUtils.open_log()

**Comment by latha-subramanian** [comment]
Latest test logs: 
[afterfix2.txt](https://github.com/PBSPro/pbspro/files/2591041/afterfix2.txt)

**Comment by bhroam** [comment]
Codacy is complaining that now that you are no longer using PBSLogUtils

**Comment by latha-subramanian** [comment]
> Codacy is complaining that now that you are no longer using PBSLogUtils

@Bhroam, Thanks for pointing that. I have made changes now. Please take a look

**Comment by latha-subramanian** [comment]
@bhroam , I have squashed and rebased for merge. All checks have passed!

**Comment by bhroam** [comment]
I merged in the PR.

**Comment by latha-subramanian** [comment]
Thanks @bhroam

**Pull request merged**